platforms/core-runtime/files/src/main/java/org/gradle/api/internal/file/archive/ZipEntry.java

```diff
@@ -61,6 +61,16 @@ enum ZipCompressionMethod {
      */
     <T> T withInputStream(IoFunction<InputStream, T> action) throws IOException;
 
+    /**
+     * Declare an action to be run against this ZipEntry's compressed content as a {@link InputStream}.
+     * The {@link InputStream} passed to the {@link IoFunction#apply(Object)} will
+     * be closed right after the action's return.
+     *
+     * This method may not be supported by all {@link ZipEntry} implementations.
+     * Use {@link #supportsRawStream()} to determine if this method can be called.
```

Comment:
🤔 What happens when the method is not supported? It would be ideal to model this with an `Optional<T>` return type, though that's not an option since `:files` must be available in workers, so no Java 8 for now. I wonder if returning `null` would be a good alternative?